spec/services/token_manager_spec.rb

```diff
@@ -0,0 +1,144 @@
+require 'rails_helper'
+
+RSpec.describe TokenManager do
+  subject{ TokenManager }
+  let(:user){ create(:user) }
+
+  describe ".generate_token" do
+    let(:token){ subject.generate_token(user.id) }
+
+    it 'generates a valid token' do
+      expect(token.split('.').count).to eq 3
+      expect{subject.decode(token)}.not_to raise_error
+    end
+
+    describe 'token expiration' do
+      it 'sets expiration time to 24hrs by default' do
+        decoded = subject.decode(token)
+        expect(decoded.first['exp']).to be_within(100).of(24.hours.from_now.to_i)
```

Comment:
Is there the need for the `be_within` matcher, as against something providing a date_range and expect the expiration date to cover a date_range. An alternative would be time_travel to a date after the token validity period and expect it to raise an error.
If you'd go the time_travel route, there's a gem `delorean` that allows something similar to this